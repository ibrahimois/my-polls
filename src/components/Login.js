import React, { useState } from "react";
import { useNavigate } from "react-router";
import { connect } from "react-redux";
import { handleSetAuthedUser } from "../actions/authedUser";
function Login({ dispatch }) {
  const navigate = useNavigate();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [seccuss, setSeccuss] = useState(true);

  const handleSubmit = (e) => {
    e.preventDefault();
    setSeccuss(!seccuss);
    dispatch(handleSetAuthedUser(username));
    if (document.referrer.includes("questions")) {
      navigate("/404");
    } else if (window.history.length > 3) {
      window.history.back();
    } else {
      navigate("/");
    }
  };

  const handleChangeUsernameChange = (e) => {
    setUsername(e.target.value);
  };

  const handleChangePasswordChange = (e) => {
    setPassword(e.target.value);
  };

  return (
    <form
      onSubmit={(e) => handleSubmit(e)}
      style={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        gap: "1rem",
      }}
    >
      <h2>Employee Polls</h2>
      <h3>Login</h3>
      <div className="options">
        <label> Username</label>
        <input
          type="text"
          placeholder="Enter username"
          onChange={(e) => handleChangeUsernameChange(e)}
          value={username}
          required
          style={{ width: "150px" }}
        />
        <label> Password</label>
        <input
          type="password"
          placeholder="Enter password"
          onChange={(e) => handleChangePasswordChange(e)}
          value={password}
          required
          style={{ width: "150px" }}
        />
      </div>
      <button
        data-testid="submit-button"
        type="submit"
        style={{ width: "150px" }}
      >
        Login
      </button>
      {seccuss === true ? (
        <div data-testid="success-header">d</div>
      ) : (
        <div data-testid="error-header">x</div>
      )}
    </form>
  );
}

export default connect()(Login);
