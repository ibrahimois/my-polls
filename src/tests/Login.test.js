import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Login from '../components/Login';
import middleware from '../middleware';
import reducer from '../reducers';
import { Provider } from 'react-redux';
import { legacy_createStore } from 'redux';
import { BrowserRouter as Router } from "react-router-dom";
const store = legacy_createStore(reducer, middleware)

const component = render(
    <Provider store={store}>
        <Router>
            <Login />
        </Router>
    </Provider>
)
const inputOne = component.getByPlaceholderText('Enter username');
const inputTwo = component.getByPlaceholderText('Enter password');
const submitBtn = component.getByTestId('submit-button');

describe('Login', () => {

    it("matches the snapshot", () => {
        expect(component).toMatchSnapshot();
    })


    it('matches the snapshot when fields are passed', () => {
        fireEvent.change(inputOne, { target: { value: 'Create a React app' } });
        fireEvent.change(inputTwo, { target: { value: 'Create a Python app' } });
        expect(component).toMatchSnapshot();
    });

})

