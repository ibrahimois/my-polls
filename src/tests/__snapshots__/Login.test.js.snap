// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        style="display: flex; flex-direction: column; justify-content: center; align-items: center; gap: 1rem;"
      >
        <h2>
          Employee Polls
        </h2>
        <h3>
          Login
        </h3>
        <div
          class="options"
        >
          <label>
             First Option
          </label>
          <input
            placeholder="Enter username"
            required=""
            style="width: 150px;"
            type="text"
            value=""
          />
          <label>
             Second Option
          </label>
          <input
            placeholder="Enter password"
            required=""
            style="width: 150px;"
            type="password"
            value=""
          />
        </div>
        <button
          data-testid="submit-button"
          style="width: 150px;"
          type="submit"
        >
          Login
        </button>
        <div
          data-testid="success-header"
        >
          d
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      style="display: flex; flex-direction: column; justify-content: center; align-items: center; gap: 1rem;"
    >
      <h2>
        Employee Polls
      </h2>
      <h3>
        Login
      </h3>
      <div
        class="options"
      >
        <label>
           First Option
        </label>
        <input
          placeholder="Enter username"
          required=""
          style="width: 150px;"
          type="text"
          value=""
        />
        <label>
           Second Option
        </label>
        <input
          placeholder="Enter password"
          required=""
          style="width: 150px;"
          type="password"
          value=""
        />
      </div>
      <button
        data-testid="submit-button"
        style="width: 150px;"
        type="submit"
      >
        Login
      </button>
      <div
        data-testid="success-header"
      >
        d
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login matches the snapshot when fields are passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
